DeratControl.Application/Users/AddEmployeeCommand.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Threading.Tasks;
+using DeratControl.Application.Interfaces;
+using DeratControl.Application.Requests;
+using DeratControl.Application.Requests.Interfaces;
+using DeratControl.Domain.Entities;
+using DeratControl.Domain.Root;
+using DeratControl.Domain.Root.Exceptions;
+using DeratControl.Domain.Root.Repositories;
+
+namespace DeratControl.Application.Users
+{
+    public sealed class AddEmployeeCommand: EmployeeDTO, IRequest
+    {     
+    }
+
+    public sealed class AddEmployeeCommandHandler : BaseCommandHandler<AddEmployeeCommand>
+    {
+        private readonly IUnitOfWork unitOfWork;
+      
+        public AddEmployeeCommandHandler(IUnitOfWork unitOfWork)
+        {
+            this.unitOfWork = unitOfWork;
+        }
+
+        protected override async Task<CommandResult> HandleRequest(CommandExecutionContext executionContext, AddEmployeeCommand request)
+        {
+            var employee = request;
+            
+            if (unitOfWork.UserRepository.Exists(employee.Email))
+                throw new UserAlreadyExistsException();
+
+            var entity = new User()
```

Comment:
what if Address/FirstName/LastName empty or null?
I'd recommend creating a constructor for User entity and initialize it with supplied arguments

In constructor check, if those args are valid